spec/requests/post_spec.rb

```diff
@@ -35,11 +35,14 @@
 
   describe 'GET index' do
     context 'when user is authenticated' do
-      let!(:user) { create(:user) }
-      let!(:posts) { create_list(:post, 6, user:) }
+      let!(:users) { create_list(:user, 3) }
+      let!(:posts_user0) { create_list(:post, 3, user: users[0]) }
+      let!(:posts_user1) { create_list(:post, 3, user: users[1]) }
+      let!(:posts_user2) { create_list(:post, 3, user: users[2]) }
 
       before do
-        sign_in user
+        sign_in users[0]
+        post relationships_path, params: { followed_id: users[1].id }
```

Comment:
As you're not actually testing the creation of relationships, you could just call the `.follow` method here